### ISSUE TITLE
Undefined Entrypoint Behaviour

### ISSUE THREAD

**Issue by senseiwells** [mcbrawls/inject#2]
The fabric inject api relies on undefined entrypoint behaviour.

The fabric mod initialisation method sets `InjectFabric.INSTANCE`, prior to this it is `null`.
```java
@Override
public void onInitialize() {
    INSTANCE = this;
    
    LOGGER.info("Inject initialising");
}
```

The examples in the README state you should hook into inject as follows:
```java
public class MyMod implements ModInitializer {
    @Override
    public void onInitialize() {
        InjectFabric.INSTANCE.registerInjector(new MyEpicHttpInjector());
    }
}
```

However [the order in which mods are initialised is undefined](https://fabricmc.net/wiki/documentation:entrypoint#a_note_about_load_order_and_phases_or_a_lack_thereof) and thus this can result in mods crashing with `NullPointerException`s as `InjectFabric.INSTANCE` may or may not be initialised yet.

A simple solution would to just have a separate instance which is statically initialised, which would guarantee that the instance is always not null.
```java
public final static InjectFabric INSTANCE = new InjectFabric();
```

**Comment by radstevee** [comment]
I think the best way to resolve this would just be to add the inject mod as a dependency in the FMJ. However I am unsure if that works with Jar-in-Jar.

**Comment by senseiwells** [comment]
No, the dependencies in the FMJ are just metadata, they are used in order to crash gracefully if a dependency is missing, it does not change the way that mods load.

There is no way to deterministically change the order in which mods load (for reasons explained in the link in my initial comment). This loading behaviour should not be relied on.

**Comment by radstevee** [comment]
Fixed in 2.2.2
